cmd/helm/template.go

```diff
@@ -0,0 +1,208 @@
+	f.StringArrayVarP(&t.renderFiles, "execute", "x", []string{}, "only execute the given templates.")
+	f.VarP(&t.valueFiles, "values", "f", "specify values in a YAML file (can specify multiple)")
+	f.StringVar(&t.namespace, "namespace", "", "namespace to install the release into")
+	f.StringArrayVar(&t.values, "set", []string{}, "set values on the command line (can specify multiple or separate values with commas: key1=val1,key2=val2)")
+	f.StringVar(&t.nameTemplate, "name-template", "", "specify template used to name the release")
+
+	return cmd
+}
+
+func (t *templateCmd) run(cmd *cobra.Command, args []string) error {
+	if len(args) < 1 {
+		return errors.New("chart is required")
+	}
+	t.chartPath = string(args[0])
+	if t.namespace == "" {
+		t.namespace = defaultNamespace()
+	}
+	// get combined values and create config
+	rawVals, err := vals(t.valueFiles, t.values)
+	if err != nil {
+		return err
+	}
+	config := &chart.Config{Raw: string(rawVals), Values: map[string]*chart.Value{}}
+
+	// If template is specified, try to run the template.
+	if t.nameTemplate != "" {
+		t.releaseName, err = generateName(t.nameTemplate)
+		if err != nil {
+			return err
+		}
+	}
+	// get abs path for top level parent chart
+	if _, err = os.Stat(t.chartPath); err == nil {
+		t.chartPath, err = filepath.Abs(t.chartPath)
+		if err != nil {
+			return err
+		}
+	}
+	// Check chart requirements to make sure all dependencies are present in /charts
+	c, err := chartutil.Load(t.chartPath)
+	if err != nil {
+		return prettyError(err)
+	}
+
+	if req, err := chartutil.LoadRequirements(c); err == nil {
+		if err := checkDependencies(c, req); err != nil {
+			return prettyError(err)
+		}
+	} else if err != chartutil.ErrRequirementsNotFound {
+		return fmt.Errorf("cannot load requirements: %v", err)
+	}
+	options := chartutil.ReleaseOptions{
+		Name:      t.releaseName,
+		Time:      timeconv.Now(),
+		Namespace: t.namespace,
+	}
+
+	err = chartutil.ProcessRequirementsEnabled(c, config)
+	if err != nil {
+		return err
+	}
+	err = chartutil.ProcessRequirementsImportValues(c)
+	if err != nil {
+		return err
+	}
+
+	// Set up engine.
+	renderer := engine.New()
+
+	vals, err := chartutil.ToRenderValues(c, config, options)
+	if err != nil {
+		return err
+	}
+
+	out, err := renderer.Render(c, vals)
+	if err != nil {
+		return err
+	}
+
+	in := func(needle string, haystack []string) bool {
+		for _, h := range haystack {
+			if h == needle {
+				return true
+			}
+		}
+		return false
+	}
+	if settings.Debug {
+		rel := &release.Release{
+			Name:      t.releaseName,
+			Chart:     c,
+			Config:    config,
+			Version:   1,
+			Namespace: t.namespace,
+			Info:      &release.Info{LastDeployed: timeconv.Timestamp(time.Now())},
+		}
+		printRelease(os.Stdout, rel)
+	}
+
+	sortedKeys := make([]string, 0, len(out))
+	for key := range out {
+		sortedKeys = append(sortedKeys, key)
+	}
+	sort.Strings(sortedKeys)
+	// If renderFiles is set, we ONLY print those.
```

Comment:
Verification of template existence is now implemented with the caveats described below.